File: CMS-SPA/src/app/core/_models/s-car-manage-record-dto.ts
import { Pagination } from "./pagination";

export class SCarManageRecordDto extends Pagination {
  licenseNumber: string;
  signInDateS: string;
  signInDateE: string;
  companyId: number;
  driverName: string;
  signInReason: string;
  departmentId: number;
  contactPerson: string;
  signOutDate:string;

  constructor() {
    super();
    this.licenseNumber = "";
    this.signInDateS = "";
    this.signInDateE = "";
    this.isPaging = true; //開分頁
    this.driverName = "";
    this.signInReason = "";
    this.contactPerson = "";
    this.signOutDate = "";  //預設全部All
  }
  public setPagination(pagination: Pagination) {
    this.currentPage = pagination.currentPage;
    this.itemsPerPage = pagination.itemsPerPage;
    this.totalItems = pagination.totalItems;
    this.totalPages = pagination.totalPages;
  }
}

File: CMS-API/DTOs/SCarManageRecordDto.cs
using System;
using API.Helpers;

namespace CMS_API.DTOs
{
    public class SCarManageRecordDto : PaginationParams
    {
        public string LicenseNumber { get; set; }
        public string SignInDateS { get; set; }
        public string SignInDateE { get; set; }
        public string CompanyId { get; set; }
        public string DriverName { get; set; }
        public string SignInReason { get; set; }
        public string DepartmentId { get; set; }
        public string ContactPerson { get; set; }
        public string SignOutDate { get; set; }

    }
}